### ISSUE TITLE
CORS errors

### ISSUE THREAD

**Issue by dmead** [auth0/wp-auth0#400]
Hi guys

I've got the plugin installed and can authenticate when i start chrome with no security (google-chrome --disable-web-security --user-data-dir)

However, when I run the plugin as normal it errors out when i use either password or passwordless logins.

In the chrome console i see:

`Failed to load https://<my-domain>.auth0.com/passwordless/verify: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://<my-domain>.com' is therefore not allowed access.`

Should I be editing my apache config to fix this? or does wordpress prefer I do something else?

**Comment by joshcanhelp** [comment]
@dmead - That error message is saying that the requested resource, `https://<my-domain>.auth0.com/passwordless/verify`, is not allowing your site to access it. I'm guessing that your dashboard is not set to allow your site to access. You want to make sure that your login page is listed under **Allowed Callback URLs** in the Client settings (that should have been set by default when it was setup).

**Comment by dmead** [comment]
Hi,

We've been wrestling with this for days now and leaning towards the idea
that it's the fault of the bitnami wordpress distro.

Are you aware of what wordpress flavor the plugin is/was tested with?
because it for sure doesn't work out of the box on bitnami.

my domain is monoclemoney.auth0.com if that helps.

**Comment by dmead** [comment]
I've done a fresh install of the latest wordpress with the latest auth0 plugin.

Still the same error. Even doing stuff like below in apache doesn't help. Gets fail on cross site policy and posts fail pre-flight checks.


`Header set Access-Control-Allow-Origin "*"`

![auth0](https://user-images.githubusercontent.com/176946/36423780-27cd7c52-160f-11e8-98df-dba891532d54.png)


Any ideas? Should this need cors adjustments at all?

**Comment by joshcanhelp** [comment]
@dmead - Apologies for the delay here. 

The CORS error is coming from the Auth0 server, it's saying that your domain is not allowed to access the Auth0 server. This is to be expected for everything except your domain, once it's been added. Just so we're on the same page, make sure that:

* Your Client settings > "Allowed Callback URLs" have the right URLs added, similar to the screenshot below (but with your domain, of course); the "Allowed Origins (CORS)" setting will pull from these:

<img width="620" alt="screenshot 2018-02-20 11 08 53" src="https://user-images.githubusercontent.com/855223/36443770-8121951c-162e-11e8-9d69-1e67c3e59f10.png">

* Make sure your Twilio account is setup properly, the SMS connection is active in your dashboard, and your client is authorized on there. 

All that said, I'm also seeing this error on my local install ("headless" WP setup similar to [this](https://davidwinter.me/install-and-manage-wordpress-with-git/), nothing out of the ordinary) so I'll need to investigate a bit to see why this is happening. It's possible that it's a configuration issue that can be corrected easily but I need to hunt down what's missing. 

One thing to note ... the newest version of Lock (the JavaScript that powers the login form) has passwordless built in so an upgrade to the plugin to account for that will be coming soon. I'll fast-track testing that and see if I can get a patch your way if the configuration doesn't pan out.

**Comment by dmead** [comment]
We have all that setup and the CORS headers configured in apache.

Twilio works because i can get an authorization code, but on submit it fails CORS policy like a POST should (pre flight blah blah).

And like I said, the entire process works if i just disable cors on the browser, so it's not the configuration.

I've even tried **patching the plugin** so it serves the javascript from my apache instance (no auth9 cdn involved) and it still errors out. If you'd like more specific debug info i'd be happy to provide privately.

**Comment by dmead** [comment]
I've tried your suggest changes and had the same result

**Comment by joshcanhelp** [comment]
Again, your Apache configuration doesn't need to change as this is a call from your browser to Auth0, won't make a difference either way. Also, where the JS loaded won't matter for this either (though there is a field in the plugin that allows you to change that). 

For the "disable cors on the browser," I'm sorry I mis-read that originally, I thought you said it **didn't** work that way. I think you're probably affected by this:

https://auth0.com/docs/cross-origin-authentication

I'll do some testing today or Monday and see if I can't help you get this figured out.

**Comment by alerdenisov** [comment]
@joshcanhelp did you some tests already?
Because same problem in my case

**Comment by joshcanhelp** [comment]
Apologies for the delayed response here. 

I'm seeing the same issue you're having @dmead and have been digging into a fix over the last week. I think the problem stems from newer tenants not having access to older API endpoints. The library that runs passwordless in this plugin is out-of-date and has since been merged with the main library that runs that login page, [Lock](https://github.com/auth0/lock). 

All that to say: this needs some work on our end and it's high priority to get that working. It will need some thorough testing but we'll make sure it's in the next release. I might be able to package it up into a plugin to try out before it and, if I'm able, I'll post instructions here. 

Thanks for your patience!

**Comment by joshcanhelp** [comment]
I put together a test of the upcoming passwordless changes in the plugin which can be downloaded and installed to get this working better on your end. Repo is here:

(joshcanhelp/auth0-pwl-beta)[https://github.com/joshcanhelp/auth0-pwl-beta]

If you're able to run `git` on the server hosting your WP instance, it's best to clone this into `wp-content/plugins/auth0-pwl-beta` so changes can be pulled in easily. 

Otherwise, you can download that as a zip and (install manually)[https://codex.wordpress.org/Managing_Plugins#Manual_Plugin_Installation]. 

Let me know if you have any issues here or, even better, if this solves the problem. If the former, happy to work with you to get it working. If the latter, it will be included in the next release. 

Thank you!

**Comment by joshcanhelp** [comment]
Good to hear! 

That error won't affect logging in, it's just trying to get the country to auto-fill values. Check your CORs setting in the dashboard to make sure you've got everything configured. 

@dmead @alerdenisov  - Are you able to test this out to see if it solves your issue?

**Comment by joshcanhelp** [comment]
Fixed in #434 and merged into `dev`. Please deactivate and delete the test plugin linked here before upgrading!

**Comment by leobarcellos** [comment]
Hi @joshcanhelp, I know this was closed and probably my comment is ~kinda~ offtopic because I'm not using wp-auth0. However I'm using auth0.js SDK and facing the same problem. Passwordless and CORS issues. 

I've configured Allowed Callback URLs already, and I'm using a custom domain. However even with the <tenant>.auth0.com I get the CORS Issue with the passwordless.verifyCode function.

Question 1: Is it possible to add the Access-Control-Allow-Origin on my custom domain? 
Question 2: Is it possible to get the auth0.min.js in my custom domain? Because I guess this issue will be resolve since calls will be in the same domain.

PS: all my calls are being requested inside the same root domain.
